app/components/ui/menu.tsx

```diff
@@ -0,0 +1,48 @@
+import { Menu, Transition } from '@headlessui/react';
+import { type ReactElement } from 'react';
+import { UserCircleIcon } from '~/components/ui/icons';
+import { Link } from '~/components/ui/links';
+import { LogoutButton } from '~/components/layout/top-nav';
+
+export function TopNavUserMenu() {
```

Comment:
As discussed, let's move this into `top-nav` as it's not a primitive UI component.